compiler/rustc_lint/src/builtin.rs

```diff
@@ -2986,6 +2986,134 @@ impl<'tcx> LateLintPass<'tcx> for AsmLabels {
     }
 }
 
+declare_lint! {
+    /// The `return_local_variable_ptr` lint detects when pointer to stack
+    /// memory associated with a local variable is returned. That pointer
+    /// is immediately dangling.
+    ///
+    /// ### Example
+    ///
+    /// ```rust,no_run
+    /// fn foo() -> *const i32 {
+    ///   let x = 42;
+    ///   &x
+    /// }
+    /// ```
+    ///
+    /// This will produce:
+    ///
+    /// ```text
+    /// error: returning a pointer to stack memory associated with a local variable
+    /// --> <source>:12:5
+    ///   |
+    /// LL|  &x
+    ///   |  ^^
+    /// ```
+    ///
+    /// ### Explanation
+    ///
+    /// Returning a pointer to memory refering to a local variable will always
+    /// end up in a dangling pointer after returning.
+    pub RETURN_LOCAL_VARIABLE_PTR,
```

Comment:
We have [guidelines for lint naming](https://rustc-dev-guide.rust-lang.org/diagnostics.html#lint-naming), under those I think it should probably be `returning_pointers_to_local_variables` or something like this.

Reply:
Thanks a lot for the pointer, this makes sense! :)